### DISCUSSION TITLE
Skin layout editor "SongProgress" alignment issue

### DISCUSSION THREAD

**Discussion by 0x84f** [ppy/osu#17536] [Q&A]
I am not sure how to describe this, but when you put the "SongProgress" module at the bottom of the screen it will not stay aligned as set up.
In the video example below I enlarged the SongProgress module and moved the progress line off-screen so only the % of progress is shown under the UR bar, it looks fine in the editor but once I get into a game it's messed up.
Might be an issue related to #17407? 
[Video example](https://streamable.com/kl16w1)

**Comment by bdach** [comment]
set the origin of song progress to bottom centre rather than top centre.

**Reply by 0x84f** [reply]
Unfortunately the issue is still persisting, [video attached](https://streamable.com/jneftr).
Full transparency, I first attempted the fix and then when I saw the issue was the same I turned on the recording.

**Reply by bdach** [reply]
i can't reproduce this behaviour. please export and attach the skin.

**Reply by 0x84f** [reply]
[Skin uploaded here, exported from lazer.](https://2127.s-ul.eu/JQItLV9f.osk)

**Reply by bdach** [reply]
i've imported the skin and it looks correct to me...

**Reply by 0x84f** [reply]
[Made a video](https://streamable.com/mnslvi) of me editing the default triangles skin, the results are even more bizarre this time.

**Reply by peppy** [reply]
Could you please drag your videos into the github windows. Please stop linking external sites.

**Reply by 0x84f** [reply]
The videos are too big to be dragged into Github, I guess I'll compress them going forwards, my bad!

**Comment by NotGumballer91** [comment]
So the inconsistent aspect ratio between the skin editor and gameplay causes that kind of misalignment. Native skin components work fine but custom skins stretches improperly?

https://user-images.githubusercontent.com/64581009/161493712-663eb4e5-e343-44cd-a7ca-ee4cc2830971.mp4

**Reply by peppy** [reply]
That's not misalignment. That's skinning working how it's intended to.

If you want a bar which always extends across the full screen width, new functionality will need to be added to support that. You are using legacy skinning, where components scale with the screen, but not *to* the screen. This is not about to change for legacy skinning.
